pkg/recordUpdate.go

```diff
@@ -4,8 +4,8 @@ package pkg
 func UpdateRecord(domain string, accessToken string, record DNSRecord, records []DNSRecord) error {
 	correctRecordExists := false
 	for _, r := range records {
-		if r.Type == record.Type {
-			if r.Name == record.Name && r.Value == record.Value+"."+domain && !correctRecordExists {
+		if r.Type == record.Type && r.Name == record.Name {
```

Comment:
we should only care about records matching the name and the type and deletion should only happen iff the record.value isn't the same as the existing value.
Previous implementation deleted all records matching the type, only exempting the record which has the existing name and value.
```go
r.Type == record.Type // e.g. filter all `A` records

r.Name == record.Name && r.Value == record.Value+"."+domain && !correctRecordExists // if it's the existing record, set correctRecordExists to true and don't update <- all other A records would be deleted here without nothing replacing them
```